app/controllers/authenticate_controller.rb

```diff
@@ -223,6 +231,35 @@ def handle_authentication_error(err)
     end
   end
 
+  def audit_authn_error(err)
+    authenticate_audit_event(err)
+  end
+
+  def audit_success
+    Audit.logger.log(
+      authenticate_audit_event(nil)
+    )
+  end
+
+  def authenticate_audit_event(err)
+    ::Audit::Event::Authn::Authenticate.new(
+      authenticator_name: authenticator_input.authenticator_name,
+      service: authenticator_input.webservice,
+      role_id: audit_role_id,
+      client_ip: authenticator_input.client_ip,
+      success: err == nil,
```

Comment:
i prefer that we have this method duplicated with `success: true` and `success: false` than checking if the err is nil.